libs/contrib/Data/Nat/Factor.idr

```diff
@@ -124,8 +123,7 @@ multOneSoleNeutral (S k) (S (S j)) prf =
         rewrite plusCommutative k j in
         rewrite sym $ plusAssociative j k (k * S j) in
         rewrite sym $ multRightSuccPlus k (S j) in
-        succInjective k (j + (S (S (j + (k * (S (S j))))))) $
-        succInjective (S k) (S (j + (S (S (j + (k * (S (S j)))))))) prf
+        injective {f = S} $ injective prf
```

Comment:
The implicit argument `{f = S}` is is required here. If it is removed, the compiler is killed, as in https://github.com/idris-lang/Idris2/issues/1630.